plugins/registry.js

```diff
@@ -377,4 +379,21 @@ export default class PluginRegistry {
     registerTranslations(getTranslationsForLocale) {
         registerPluginTranslationsSource(this.id, getTranslationsForLocale);
     }
+
+    registerRightHandSidebarComponent(component, title) {
+        const id = generateId();
+
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_COMPONENT,
+            name: 'RightHandSidebarComponent',
+            data: {
+                id,
+                pluginId: this.id,
+                component,
+                title,
+            },
+        });
+
+        return {id, showRHSPlug};
```

Comment:
I'm not sure if we do this sort of thing elsewhere, but I like the idea of returning the action that can be used to show the component.

Would it make sense to pass the plugin ID for the user? We could return something like `showRHSPlug: () => showRHSPlug(id)` so that the plugin can call the action without needing to pass in its own plugin ID as well. That would prevent one plugin from showing another's RHS component, but that's probably fine. We could always do something with default arguments if we decide that we want that behaviour.

Reply:
Yes, I think this is a good improvement because it is more secure as you mention and keeps up code on the plugin side to get their own id from the manifest.